src/megaapi_impl.cpp

```diff
@@ -10938,13 +11023,13 @@ void MegaApiImpl::sendPendingTransfers()
                     break;
                 }
 
-                if (!transfer->isStreamingTransfer() && node && node->type != FILENODE)
+                if (!transfer->isStreamingTransfer() && ((node && node->type != FILENODE) || (publicNode && publicNode->getType() != FILENODE)) )
                 {
                     // Folder download
                     transferMap[nextTag] = transfer;
                     transfer->setTag(nextTag);
                     MegaFolderDownloadController *downloader = new MegaFolderDownloadController(this, transfer);
-                    downloader->start();
+                    downloader->start(publicNode);
```

Comment:
publicNode is not only set in the case of public nodes. It's set also for the own nodes. I have changed the name of the variable to show it.